firmware/application/apps/ui_flash_utility.cpp

```diff
@@ -30,6 +30,30 @@ static const char16_t* firmware_folder = u"/FIRMWARE";
 Thread* FlashUtilityView::thread{nullptr};
 static constexpr size_t max_filename_length = 26;
 
+bool valid_firmware_file(std::filesystem::path::string_type path) {
+    File firmware_file;
+    uint32_t read_buffer[128];
```

Comment:
Shouldn't we initialize them to 0 ?

Reply:
the read_buffer and checksum ?

Reply:
Thanks for noticing that.  I have just fixed the check after the file read operation to make sure that the buffer has been filled completely before doing the checksum test.

I initialized the checksum to 1 because I wanted the checksum test to fail if the file cannot be opened.